sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/HttpRetryPolicy.java

```diff
@@ -37,7 +37,7 @@
 /**
  * A pipeline policy that retries when a recoverable HTTP error or exception occurs.
  */
-public class HttpRetryPolicy implements HttpPipelinePolicy {
+public final class HttpRetryPolicy implements HttpPipelinePolicy {
```

Comment:
asking here, but it applies to instr and redirect policy too - wouldn't we want to allow custom retry/redirect/instr policy implementations? why?

Reply:
Thoughts on adding hooks into redirect, retry, and instrumentation that would allow for customization without needing to extend the entire policy? This would let us maintain a general behavior but allow for properly vetted and safe mutations to behavior.

Reply:
To be done in a follow-up PR